self-hosting/deployment.yaml

```diff
@@ -0,0 +1,143 @@
+        - name: omnivore-api
+          image: ghcr.io/lawrencegripper/omnivore:17-10-23-11
+          imagePullPolicy: Always
+          ports:
+            - containerPort: 8080
+          env:
+            - name: PG_DB
+              value: omnivore
+            - name: PG_PASSWORD
+              value: some-postgres-password
+            - name: POSTGRES_DB
+              value: omnivore
+            - name: POSTGRES_PASSWORD
+              value: some-postgres-password
+            - name: POSTGRES_USER
+              value: omnivore
+            - name: POSTGRES_HOST
+              value: localhost
+            - name: PG_USER
+              value: omnivore
+            - name: PG_HOST
+              value: localhost
+            - name: PG_PORT
+              value: "5432"
+            - name: PG_POOL_MAX
+              value: "20"
+            - name: API_ENV
+              value: local
+            - name: JAEGER_HOST
+              value: "jaeger"
+            - name: REMINDER_TASK_HANDLER_URL
+              value: /svc/reminders/trigger
+            - name: CLIENT_URL
+              value: http://omnivore-web.some-tailnet.ts.net
+            - name: GATEWAY_URL
+              value: http://localhost:8080/api
+            - name: ELASTIC_URL
+              value: http://localhost:9200
+            - name: JWT_SECRET
+              value: 2f6f53a8-6626-477d-af94-801f652e9919
+            - name: SSO_JWT_SECRET
+              value: 2f6f53a8-6626-477d-af94-801f652e9919
+          volumeMounts:
+            - mountPath: /config
+              name: config
+        - name: postgresql
+          image: ankane/pgvector
+          ports:
+            - containerPort: 5432
+          env:
+            - name: POSTGRES_DB
+              value: omnivore
+            - name: POSTGRES_PASSWORD
+              value: some-postgres-password
+            - name: POSTGRES_USER
+              value: omnivore
+            - name: PG_POOL
+              value: "20"
+          lifecycle:
+            preStop:
+              exec:
+                command: ["/usr/local/bin/pg_ctl stop -D /var/lib/postgresql/data -w -t 60 -m fast"]
+          resources:
+            requests:
+              cpu: 100m
+              memory: 100Mi
+            limits:
+              cpu: 250m
+              memory: 600Mi
+          volumeMounts:
+            - mountPath: /var/lib/postgresql/data/
+              name: database
+        - name: elasticsearch
+          image: docker.elastic.co/elasticsearch/elasticsearch:8.10.4
+          ports:
+            - containerPort: 9200
+          env:
+            - name: ELASTIC_PASSWORD
+              value: some-postgres-passwordelastic
+            - name: discovery.type
+              value: single-node
+          volumeMounts:
+            - mountPath: /usr/share/elasticsearch/data
+              name: elastic-data
+            # - mountPath: /usr/share/elasticsearch/config
+            #   name: elastic-config
```

Comment:
imo we shouldn't be deploying Elasticsearch as a sidecar to this Deployment (or let alone as a Deployment instead of StatefulSet). I think the way to go would be to make a stand-alone Omnivore Helm Chart which has a dependency on some Elasticsearch Helm Chart. Either that or we leave it up to the user to manage / have an Elasticsearch instance deployed.

Reply:
The same for pg too. Lost of self-hoster already installed cloudnative-pg/Zalando PG or something similar. Also, splitting web, api and contect-fetch is more k8s way. But this commit is a good start, good job! Thanks for your work.